contracts/src/access/witnesses/interface.ts

```diff
@@ -1,13 +1,8 @@
 import type { WitnessContext } from '@midnight-ntwrk/compact-runtime';
-import type {
-  Maybe,
-  MerkleTreePath,
-} from '@openzeppelin/midnight-apps-compact-std';
-import type {
-  AccessControl_Role,
-  Ledger,
-} from '../artifacts/Index/contract/index.cjs';
-
+import type { Role } from '../../../artifacts/access/Index/contract/index.cjs';
+// TODO: (team): I noticed that we are using the mock contract's ledger type here, but I am not sure if this is correct way.
```

Comment:
Agreed. The `Ledger` type should be generic in both the interface and default impl

Reply:
We need to update both the contracts and the sim

Reply:
That basically will be solved if we follow that pattern proposed in this commit: https://github.com/OpenZeppelin/midnight-apps/pull/275/changes/e180fa9d782afbd7629af0855504e44f219b53a1